api/read.yaml

```diff
@@ -0,0 +1,97 @@
+openapi: 3.0.1
+info:
+  title: Ampersand public read API
+  version: 1.0.0
+servers:
+  - url: https://read.withampersand.com/v1
+paths:
+  /projects/{projectIdOrName}/integrations/{integrationId}/objects/{objectName}:
+    post:
+      summary: Create read operation for records
+      operationId: createRead
+      tags: ["Read"]
+      parameters:
+        - name: projectIdOrName
+          in: path
+          required: true
+          schema:
+            type: string
+        - name: integrationId
+          in: path
+          required: true
+          schema:
+            type: string
+        - name: objectName
+          in: path
+          required: true
+          schema:
+            type: string
+        - name: X-Amp-Request-Type
+          in: header
+          required: true
+          schema:
+            type: string
+            enum: [read]
+      requestBody:
+        description: Read Request
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/CreateReadAsync"
+      responses:
+        201:
+          description: The success response
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ReadResultAsync"
+        400:
+          description: The failure response for bad request.
+          content:
+            application/json:
+              schema:
+                $ref: "../problem/problem.yaml#/components/schemas/ApiProblem"
+        500:
+          description: Internal server error
+          content:
+            application/json:
+              schema:
+                $ref: "../problem/problem.yaml#/components/schemas/ApiProblem"
+        default:
+          description: Error
+          content:
+            application/problem+json:
+              schema:
+                $ref: "../problem/problem.yaml#/components/schemas/ApiProblem"
+
+components:
+  schemas:
+    CreateReadAsync:
```

Comment:
This should be named `ReadRequest` because we will also support synchronous reads in the future, so it shouldn't be named `CreateReadAsync`